lib/redmine/menu_manager/top_menu_helper.rb

```diff
@@ -60,8 +64,13 @@ def render_logo
   end
 
   def render_logo_icon
-    mode_class = User.current.pref.theme === "dark" ? "op-logo-icon_dark" : "op-logo-icon"
-    render Primer::BaseComponent.new(tag: :div, classes: mode_class)
+    mode_class = "op-logo-icon_white" unless User.current.pref.high_contrast_theme?
+    link_to(I18n.t("label_home"), configurable_home_url, class: "op-logo op-logo-icon op-logo--link #{mode_class}")
```

Comment:
There's no need to do string concatenation here. The `class` option is passed to [`token_list`](https://api.rubyonrails.org/classes/ActionView/Helpers/TagHelper.html#method-i-token_list) (`class_names` under-the-hood:

```suggestion
    link_to(I18n.t("label_home"), configurable_home_url, class: ["op-logo", "op-logo--link", mode_class])
```

you could even inline `mode_class`:

```suggestion
    link_to(I18n.t("label_home"), configurable_home_url, class: ["op-logo", "op-logo--link", { "op-logo-icon_white": dark_mode?, "op-logo-icon": !dark_mode? ])
```